owncloudApp/src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.kt

```diff
@@ -0,0 +1,467 @@
+            isEnabled = false
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.action_share_file -> {
+                mContainerActivity.fileOperationsHelper.showShareFile(file)
+                true
+            }
+            R.id.action_open_file_with -> {
+                mContainerActivity.fileOperationsHelper.openFile(file)
+                true
+            }
+            R.id.action_remove_file -> {
+                val dialog = RemoveFilesDialogFragment.newInstance(file)
+                dialog.show(parentFragmentManager, FTAG_CONFIRMATION)
+                true
+            }
+            R.id.action_rename_file -> {
+                val dialog = RenameFileDialogFragment.newInstance(file)
+                dialog.show(parentFragmentManager, FTAG_RENAME_FILE)
+                true
+            }
+            R.id.action_cancel_sync -> {
+                (mContainerActivity as FileDisplayActivity).cancelTransference(file)
+                true
+            }
+            R.id.action_download_file, R.id.action_sync_file -> {
+                mContainerActivity.fileOperationsHelper.syncFile(file)
+                startObservingProgressForDownload()
+                true
+            }
+            R.id.action_send_file -> {
+
+                // Obtain the file
+                if (!file.isAvailableLocally) {  // Download the file
+                    Timber.d("%s : File must be downloaded", file.remotePath)
+                    (mContainerActivity as FileDisplayActivity).startDownloadForSending(file)
+                } else {
+                    mContainerActivity.fileOperationsHelper.sendDownloadedFile(file)
+                }
+                true
+            }
+            R.id.action_set_available_offline -> {
+                mContainerActivity.fileOperationsHelper.toggleAvailableOffline(file, true)
+                true
+            }
+            R.id.action_unset_available_offline -> {
+                mContainerActivity.fileOperationsHelper.toggleAvailableOffline(file, false)
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    /**
+     * Updates the view with all relevant details about that file.
+     *
+     * @param forcedTransferring Flag signaling if the file should be considered as downloading or uploading,
+     * although [FileDetailsViewModel.isDownloadPending] and [FileUploaderBinder.isUploading] return false.
+     * @param refresh            If 'true', try to refresh the whole file from the database
+     */
+    private fun updateFileDetails(forcedTransferring: Boolean, refresh: Boolean) {
+        val storageManager = mContainerActivity.storageManager
+        if (refresh && storageManager != null) {
+            file = storageManager.getFileByPath(file.remotePath)
+        }
+        val file = file
```

Comment:
🤔

Reply:
It was written that way in Java, I suppose it prevents mutability 🤔